### PR TITLE
Added documentation for material mixing

### PR DESCRIPTION
**Pull request by SamPUG**
This pull request contains changes related to #1506 - Write documentation for the new material mixing feature added in #1447 

I have added a a section in the user guide which explains the new mix_materials method. I have made sure to note that fractions must sum to one unless volume fractions are specified. I have also included a warning that you can't use materials which contain thermal scattering data. I think these are the main limitations of the method.

I also added to the pincell example Jupyter notebook with an example of how to create mox by mixing a previously created uo2 material and a newly created puo2 material. I think the fractions I used for puo2 and making the MOX are sensible.

To test the additions, I have looked at a preview of the user guide using a VSCode plugin. I used jupyter notebook to make the edits to the example and re-ran it to test it works with the addition.

### CODE REVIEW

**Comment by simondrichards** [comment]
Just to note: the atomic and weight fractions must sum to **one** (not zero as stated in the previous comment). This is described correctly in the updates to the users guide and Jupyter notebook, so it's just a typo on the comment. 

I have also built the html users guide and the addition looks fine there.

**Comment by SamPUG** [comment]
> Just to note: the atomic and weight fractions must sum to **one** (not zero as stated in the previous comment). This is described correctly in the updates to the users guide and Jupyter notebook, so it's just a typo on the comment.

Thanks @simondrichards, I have corrected the initial comment.

File: docs/source/usersguide/materials.rst [hunk 1]
```diff
@@ -141,6 +141,33 @@ attribute, e.g.,
...
 
+-----------------
+Material Mixtures
+-----------------
+
+In OpenMC it is possible to mix any number of materials to create a new material
+with the correct nuclide composition and density. The 
+:meth:`Material.mix_materials` method takes a list of materials and
+a list of their mixing fractions. Mixing fractions can be provided as atomic 
+fractions, weight fractions or volume fractions. The fraction type
```
Comment (paulromano) [inline docs/source/usersguide/materials.rst]:
```suggestion
fractions, weight fractions, or volume fractions. The fraction type
```

File: docs/source/usersguide/materials.rst [hunk 2]
```diff
@@ -141,6 +141,33 @@ attribute, e.g.,
...
+is specifed by adding 'ao', 'wo' or 'vo' respectively as the third argument.
```
Comment (paulromano) [inline docs/source/usersguide/materials.rst]:
```suggestion
can be specified by passing 'ao', 'wo', or 'vo' as the third argument, respectively. 
```

File: examples/jupyter/pincell.ipynb [hunk 3]
```diff
@@ -435,6 +435,39 @@
...
+    "\n",
+    "# Create the mixture\n",
+    "mox = openmc.Material.mix_materials([uo2, puo2], [0.97, 0.03], 'wo')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The 'wo' argument in the `mix_materials()` method specifies that the fractions are weight fractions. Materials can also be mixed by atomic and volume fractions with 'ao' and 'vo' respectively. For 'ao' and 'wo' the fractions must sum to one. For 'vo' if fractions do not sum to one, the remaining fraction is set as void."
```
Comment (paulromano) [inline examples/jupyter/pincell.ipynb]:
```suggestion
    "The 'wo' argument in the `mix_materials()` method specifies that the fractions are weight fractions. Materials can also be mixed by atomic and volume fractions with 'ao' and 'vo', respectively. For 'ao' and 'wo' the fractions must sum to one. For 'vo', if fractions do not sum to one, the remaining fraction is set as void."
```

**Pull request merged**